### PR TITLE
Add eslint with mocha and prettier, markdownlint and yaml-lint

### PR DESCRIPTION
**Pull request by rachidlamouri**
I recommend reviewing the individual commits.

All styling rules should be configured in `.prettierrc.js` since it has better logic for auto fixing rules and the prettier plugin disables conflicting eslint rules. I included docs for updating vscode to format on save.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -17,7 +17,18 @@ jobs:
           node-version: '16.x'
       - name: Install
         run: npm ci
+      - name: Lint Code
+        if: always()
```
Comment (jshemas) [inline .github/workflows/ci.yaml:21]: Are these `if`s needed?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -18,17 +18,18 @@ jobs:
       - name: Install
         run: npm ci
       - name: Lint Code
-        if: always()
-        run: npm run lint:ts
+        # if: always()
+        # run: npm run lint:ts
+        run: exit 1
```
Comment (jshemas) [inline .github/workflows/ci.yaml:23]: `exit 1` causes the build to fail.

Comment (rachidlamouri) [reply .github/workflows/ci.yaml:23]: won't fix

Comment (rachidlamouri) [reply .github/workflows/ci.yaml]: changed to exit 2

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (rachidlamouri) [reply .github/workflows/ci.yaml:21]: They let everything run even if one node fails. It'll still fail the check on a single failure

**Review by jshemas — APPROVED** [review]

**Pull request merged**